master/internal/api_project_intg_test.go

```diff
@@ -0,0 +1,244 @@
+//go:build integration
+// +build integration
+
+package internal
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+
+	"github.com/determined-ai/determined/master/internal/mocks"
+	"github.com/determined-ai/determined/master/internal/project"
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/proto/pkg/apiv1"
+	"github.com/determined-ai/determined/proto/pkg/projectv1"
+)
+
+var projectAuthZ *mocks.ProjectAuthZ
```

Comment:
I don't see much value in testing at the basic auth interface like ```CanGetProject``` methods since they should be mostly covered by already existing e2e tests.

I do think that will change for RBAC in EE.

I personally feel that the integration tests with this mock interface is useful to make sure that the interface is being called correctly.